### PR TITLE
VACMS-6449 Official and Common name field treatments on VC's

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
Relates to #6449

## Testing done
Visual / behat

## Screenshots
image
image
image
image

## QA steps
 - [x] As an administrator, go to [node/3634/edit](https://pr7045-ayevcjbsa1kkqevl8kletojoj6dsnaue.ci.cms.va.gov/node/3634/edit) and visually verify you see Name of Vet Center (field_official_name)  field as disabled, with tooltip indicator icon that appears on hover.
 - [x] As an administrator, go to [node/3840/edit](https://pr7045-ayevcjbsa1kkqevl8kletojoj6dsnaue.ci.cms.va.gov/node/3634/edit) and visually verify presence of Common name field (title) and that you can edit it.
 - [x] Change the value in Common name field, and confirm it is reflected in the page title on the view page and the node form.
 - [x] Login as any user who is not an admin and has access to node 3840, and go to `node/3840/edit` and visually verify absence of the Official name field.
 - Go back to the node as admin... make the tow names different and save-publish the node
 - [x] Login as any user who is not an admin and has access to node 3840, and go to `node/3840/edit` and visually verify  bpth title fields appear and they are both locked.
 - [x] Go to `admin/structure/types/manage/vet_center/form-display` and visually verify Tooltip textfield formatter matches screenshots, above.

**OPTIONAL EXTRA CREDIT TEST** 
 - [ ] Pull branch down locally, comment out line 212 of `docroot/modules/custom/va_gov_consumers/va_gov_consumers.module` and visit any vet center.
 - [ ] Change Page title field and save.
 - [ ] Visually verify that Common name field now appears (see second screenshot)

### CODE REVIEW

**Review by netboss — DISMISSED** [review]
@ethanteague Looking good!

**Comment by EWashb** [comment]
Screen Shot 2021-11-23 at 3 02 03 PM
@ethanteague there are a few things on node/3634/edit 

There's some padding around the tooltip icon, can that be removed? Also the help text should be below the Page title. Otherwise that section looks good. Checking the extra credit stuff now!

**Comment by ethanteague** [comment]
> Screen Shot 2021-11-23 at 3 02 03 PM
> 
> @ethanteague there are a few things on node/3634/edit
> There's some padding around the tooltip icon, can that be removed? Also the help text should be below the Page title. Otherwise that section looks good. Checking the extra credit stuff now!

Thanks for the quick review / feedback, @EWashb :heart: 
Updated: 
image

**Review by kevwalsh — COMMENTED** [review]
<details><summary>Non-admin experience</summary>


Vet_Centers_forms_–_Figma

</details>


<details><summary>Admin experience</summary>

Vet_Centers_forms_–_Figma


</details>

First off, i think we missed an engineering review of the designs somehow. Or maybe just i missed it.  This has led to confusing ACs (my bad). And now i'm doing some design review in an engineering PR. But at least we're all here together, right? 😜 

1. The QA instructions for this PR don't cover the two scenarios (Scenario 1 for Prescott vs Scenario 2 for Raleigh). Scenario 1 in the design is somewhat mislabeled, technically. All Vet Centers have a Page title and a Common name stored in the database.  In most cases, those are the same, Scenario 1 should apply in rare cases _when they are not the same_ (eg "Prescott Vet Center" vs "Dr Cameron Prescott..." in [3785](https://pr7045-ayevcjbsa1kkqevl8kletojoj6dsnaue.ci.cms.va.gov/node/3785/edit)). Scenario 2 is 99% of cases, like [Nassau](https://pr7045-ayevcjbsa1kkqevl8kletojoj6dsnaue.ci.cms.va.gov/node/3634/edit).
2. Moving the help text for a field above the field is probably its own issue. @msbtterswrth and @ethanteague have an open PR to make this change for all textarea (multiline) text fields. I have a sense that we want to do this for single line textfields too, but i think it should go into its own issue, and we can probably do this site-wide, not just for this one field. @EWashb if you feel that help text placement is critical to this design, we can look at one-off behavior here, but this is desirable for all single line text fields and i think it should be done system-wide. 
3. The admin experience should be different than the non-admin experience (see 2nd screenshot). But i'm a bit concerned about the different help text for admins. I think it's fine if we provide the same help text for both cases. "This Vet Center also has a more commonly used name, which will be used everywhere on the site except for the page title."

File: config/sync/node_link_report.settings.yml [hunk 1]
```diff
@@ -12,6 +12,7 @@ domains_to_skip: |-
   https://imablefoundation.org/
   https://teampossabilities.org/
   https://woundedwarriorsmammoth.org/
+  https://www.spiritrock.org
```
Comment (kevwalsh) [inline config/sync/node_link_report.settings.yml:15]: is this intentional?  i imagine so but just in case.

**Comment by kevwalsh** [comment]
OK, so Ethan beat me to it with the help text change 🤣 ... but my concern still stands, this should probably be a site-wide change and one-offs become tech debt. @ethanteague do you have a take on that?  Looks like you did it with some js magic. I'm cool with this (well, defer to @swirtSJW), as long as when we address this site-wide for single line text fields, we go back and rip this out. I'll create an issue for that.

**Comment by kevwalsh** [comment]
I created #7048 to move the help text for single line text fields, across the site.

**Comment by kevwalsh** [comment]
It just occurred to me that we want to _always_ show the Common name field, editable, for admins, even when it is identical. Because that's how admins will override the Official name. /cc @swirtSJW

File: config/sync/node_link_report.settings.yml (hunk 1 above)
Comment (ethanteague) [reply config/sync/node_link_report.settings.yml:15]: Lol, no! @swirtSJW .......any idea how / why that ended up in here when I exported?

**Comment by ethanteague** [comment]
> OK, so Ethan beat me to it with the help text change rofl ... but my concern still stands, this should probably be a site-wide change and one-offs become tech debt. @ethanteague do you have a take on that? Looks like you did it with some js magic. I'm cool with this (well, defer to @swirtSJW), as long as when we address this site-wide for single line text fields, we go back and rip this out. I'll create an issue for that.

@swirtSJW @kevwalsh - I'm good with leaving this in for now (no biggie to augment / remove) = I've written these couple of js functions with the plan being to use them for all instances for all field types. I think we need to go the js route because field label and field help text live in two separate templates, and require two separate preprocess functions. Not to mention the handful of custom form field templates we already have....would end up being a lot of custom code to go the non-js way, and imho, open up the door for lots of different points of failure.

**Comment by ethanteague** [comment]
> It just occurred to me that we want to _always_ show the Common name field, editable, for admins, even when it is identical. Because that's how admins will override the Official name. /cc @swirtSJW

Great thinking. Will do this.

File: config/sync/node_link_report.settings.yml (hunk 1 above)
Comment (swirtSJW) [reply config/sync/node_link_report.settings.yml:15]: node link report items are config ignored, but drupal only ignores them on the way in.   So whenever there are things added to the UI, they will bubble out on config export.   It is kind of expected that we will see them come out any time there is a config change to the ignore list.

**Comment by EWashb** [comment]
@kevwalsh oops! yea I think I'm confused about when/where to review. It could be helpful to go over that with me and maybe the entire design team so we are doing the right things that the right time. Apologies for the confusion!

About the text placement- I agree that it's a bigger issue. Here doesn't seem nearly as critical as other areas! I was having an issue with my tugboat yesterday so was unable to do any other reviewing, but maybe that's a good thing 😆

**Comment by kevwalsh** [comment]
@EWashb the lack of engineering review of designs is on me. We have so many adjacent epics that i've admittedly lost track of which ones have gone through what steps. 

It turns out Ethan's approach to the moving the help text _is_ the scaleable solution (sorry buddy, i just learned that via #7011). so i'm a bit less concerned about it than i expressed last night.

**Comment by ethanteague** [comment]
Okay @kevwalsh @EWashb : pulled out the help text moving logic in this pr, as universal help text logic is being handled by https://github.com/department-of-veterans-affairs/va.gov-cms/pull/7066

Also closed down https://github.com/department-of-veterans-affairs/va.gov-cms/pull/6961 as the work being performed in this pr renders 6961 obsolete.
cc: @swirtSJW

**Review by EWashb — COMMENTED** [review]
I hope you weren't waiting on me here @ethanteague! This is looking good and I have no other feedback, especially since you're awesome and figured out how to move the help text.

**Comment by EWashb** [comment]
Screen Shot 2021-11-30 at 1 54 53 PM

Oops...one thing. apologies! After you do change the name and check the work, it did save the grey padding here.

**Comment by ethanteague** [comment]
> Screen Shot 2021-11-30 at 1 54 53 PM
> 
> Oops...one thing. apologies! After you do change the name and check the work, it did save the grey padding here.

G

> Screen Shot 2021-11-30 at 1 54 53 PM
> 
> Oops...one thing. apologies! After you do change the name and check the work, it did save the grey padding here.

Glad you caught that, @EWashb . That's a keen eye :heart: . Will fix.

**Review by EWashb — COMMENTED** [review]
I surprised myself @ethanteague! Thanks for making that change. All looks good from my perspective.

**Review by swirtSJW — COMMENTED** [review]
Bahh  Sorry I just realized I never finalized my pending comments.  I made them a few days ago..  @ethanteague Lets talk this though on a zoom to clear up why we are using a different method for tool tips.

File: docroot/modules/custom/va_gov_consumers/js/textFieldTooltips.es6.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        nameButtonContainerDiv.appendChild(nameButton);
+        document
+          .getElementById(`edit-${placementTarget}`)
+          .after(nameButtonContainerDiv);
+      };
+
+      const targets = [
+        {
+          text:
+            "Why can’t I edit this?\nThis content is automatically populated from centralized databases, and helps maintain consistent information across all of VA.gov.",
```
[the remark this answers is not in the document]
Comment (ethanteague) [reply docroot/modules/custom/va_gov_consumers/js/textFieldTooltips.es6.js:33]: Adding a comment here so we remember what we are doing / talked about: Path forward is to create a new fieldformatter for text fields with tooltips. Also want to move our fieldgroup formatters that use tooltips over to the tippy-css version I'm using in this pull request.

**Review by EWashb — COMMENTED** [review]
All the UX updates looked good to me!

**Review by swirtSJW — APPROVED** [review]
Looks great.   Most excellent work @ethanteague

**Pull request merged**